app/javascript/packs/application.js

```diff
@@ -14,6 +14,11 @@ import 'javascripts/jquery.easyModal';
 import Rails from 'rails-ujs';
 Rails.start();
 
+// i18n-js
+import * as I18n from 'i18n-js';
+window.I18n = I18n;
+require('translations');
```

Comment:
I was missing the require translations part, nice.
(but it breaks the build and I'm working to understand why)

Reply:
@david-yz-liu found the problem, it's working locally for you because you kept the i18n-js gem, without it we need to find a way to generate that public/javascripts/translations.js
This is someone who made it work, but in a really complicated way: https://engineering.klarna.com/migrating-from-rails-asset-pipeline-to-node-s-webpack-684230e3a93a